crates/core/src/db/relational_db.rs

```diff
@@ -329,6 +342,21 @@ impl RelationalDB {
                 }),
             };
 
+            // NOTE: Ignoring empty transactions.
+            //
+            // Technically, this is incorrect, because transactions can have
+            // side-effects. Namely, they can:
+            //
+            // - write to the reducer log
+            //   (sometimes used to assert that a reducer indeed ran)
+            // - schedule a reducer to run at a later time
+            // - potentially in the future: perform RPCs
```

Comment:
Not sure what the first one is but in the near future the second and third ones will write to the log. Writing to the log for RPCs is necessary to ensure exactly once delivery (because in the event of a crash you need to be able to retransmit those messages for which you had not yet received an ack)

Reply:
Actually, I meant to change the logic to ignore all transactions which do not mutate the database. But, that is wrong:

A module may not define `connect` / `disconnect`, or only define one of them. Or define them, but not modify any tables. Or worse, return an error.

I don't know what to do in the last case, but ignoring that we do indeed need to write out transactions which originate from reducer calls but don't modify the database.